### PR TITLE
Notion #27 - Mirror issues

### PR DESCRIPTION
**Pull request by EnzoGrn**
### Description
Add a issue linker between this repository and the remote repository from Epitech.
This linker copies each issue and its comments from this repository to the remote repository.
The copy has a reference to the original issue.

### Related Issue(s)
[Notion #27](https://www.notion.so/Migrer-sous-github-project-mirroring-10e55ef261d880218a81fdce27ff32b2?pvs=4)

### Changes Made
Added a new GitHub Action:
- Created a GitHub Action to automatically link issues between this repository and the remote Epitech repository.
- The action triggers when a new issue or comment is created or updated.

Implemented issue and comment synchronization:
- Developed functionality to copy issues and their associated comments from this repository to the remote repository.
- Ensured each copied issue includes a reference back to the original issue in this repository.

Added issue-reference mapping:
- Introduced a mapping system to track the relationship between original and copied issues, ensuring accurate synchronization between repositories.

### Testing
Test in two custom repositories, and trying to copy issue and comments.

### Screenshots (if applicable)

image
image
image

### Checklist
- [x] I have tested these changes thoroughly.
- [x] The code follows the project's style guide and coding conventions.
- [x] I have updated the relevant documentation (if applicable).
- [x] All tests passed successfully.
- [x] I have checked for any potential conflicts with other branches.
- [x] I have added secrets variables

### Definition of Done
- Issues are copied in the remote repository
- Comments of issue are copied in the remote repository
- All workflows steps past the test and run successfully

### CODE REVIEW

File: .github/workflows/link_issue.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: Link Issues Across Repositories
+
+on:
+  workflow_dispatch:
+  issues:
+    types: [opened, edited]
```
Comment (PharaEthan) [inline .github/workflows/link_issue.yml]: Si une issues ce 'closed' sa ne la close dans l'autre repos ducoup ?

File: .github/workflows/link_issue.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Check out the repository
+        uses: actions/checkout@v2
+
+      - name: Display the GitHub event payload
+        run: |
+          cat "$GITHUB_EVENT_PATH"
+      
+      - name: Debug API Request
+        run: |
+          response=$(curl -s -o response.json -w "%{http_code}" -H "Authorization: token ${{ secrets.PAT }}" "https://api.github.com/repos/EnzoGrn/autorelease/issues")
```
Comment (PharaEthan) [inline .github/workflows/link_issue.yml]: 'https://api.github.com/repos/EnzoGrn/autorelease/issues', c'est la bonne adresse ?

File: .github/workflows/link_issue.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        run: |
+          echo "Title: ${{ env.title }}"
+          echo "Body: ${{ env.body }}"
+          echo "Number: ${{ env.number }}"
+
+      - name: Check if issue already exists
+        id: check_issue
+        run: |
+          existing_issues=$(curl -s -H "Authorization: token ${{ secrets.PAT }}" \
+          "https://api.github.com/repos/${{ env.to_repo }}/issues?state=open&filter=all")
```
Comment (PharaEthan) [inline .github/workflows/link_issue.yml]: Ici tu met bien un ${{ env.to_repo }}, ce qui veux dire que dans celui plus haut tu doit mettre ton répertoire d'origine, je crois et donc pas autorelease mais Vermines

**Review by PharaEthan — CHANGES_REQUESTED** [review]

File: .github/workflows/link_issue.yml (hunk 1 above)
Comment (EnzoGrn) [reply .github/workflows/link_issue.yml]: J'ai ajouté cette fonctionnalité, ainsi que la réouverture

File: .github/workflows/link_issue.yml (hunk 2 above)
Comment (EnzoGrn) [reply .github/workflows/link_issue.yml]: C'est modifié

File: .github/workflows/link_issue.yml (hunk 3 above)
Comment (EnzoGrn) [reply .github/workflows/link_issue.yml]: Yes c'est changé

**Review by PharaEthan — APPROVED** [review]

**Pull request merged**